pkg/globalnet/controllers/global_ingressip_controller.go

```diff
@@ -148,12 +177,56 @@ func (c *globalIngressIPController) onDelete(ingressIP *submarinerv1.GlobalIngre
 
 	key, _ := cache.MetaNamespaceKeyFunc(ingressIP)
 
+	if ingressIP.Spec.ServiceRef != nil {
+		chainName := ingressIP.GetAnnotations()[kubeProxyIPTableChainAnnotation]
+		if chainName == "" {
+			klog.Warningf("%q annotation is missing on %q", kubeProxyIPTableChainAnnotation, key)
+			return nil, true
+		}
+
+		err := c.iptIface.RemoveIngressRulesForService(ingressIP.Status.AllocatedIP, chainName)
+		if err != nil {
+			klog.Errorf("Error while deleting Service %q ingress rules. %v", key, err)
+			return nil, true
```

Comment:
This will elide releasing the IP on L194. Perhaps we should move this block below L194 to ensure we always release the IP?

Reply:
I removed the return values if there is a failure. Also, we want to delete the service rules first before we release the IP to the pool.